### DISCUSSION TITLE
SSG i18n 404 pages.

### DISCUSSION THREAD

**Discussion by susickypavel** [vercel/vercel#8664] [Help] [answered]
Hi 👋,

It seems like there is no optimal way of doing internationalized pages using SSG.

```
fr/
├─ index.html
├─ 404.html
index.html
404.html
```

Vercel returns only the 404 page located in the root, even for other languages, which is not the desired behavior.

We can use `vercel.json` rewrites to map the path to the correct 404 page, but this makes the page return 2xx status code which is not optimal.

The solutions I currently came up with are either to:

1. Use `vercel.json` rewrite with API route returning 404 page and correct status code.
2. Use `vercel.json` rewrite with indexing disabled on 404 pages (using HTTP header or meta tag).

Is there a more feasible solution that I'm missing? 

Usage of `vercel.json` routes is not possible in most cases as this can't be combined with `redirect`, `cleanUrls` etc.

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @Thesoreon! Your [custom 404 page](https://vercel.com/guides/custom-404-page) solution will depend on what framework or library you are using for your app. You could use a custom 404 API response as a localization technique to get the error message in the correct language for your visitor. Check out the section about [creating a custom 404 API response](https://vercel.com/guides/custom-404-page#what-if-i-need-a-custom-404-api-response) for more information.

**Reply by susickypavel** [reply]
Yeah, it seems like only reasonable solution in my use case. Thanks for answering :)
